### PR TITLE
feat: allow progress bar while batching

### PR DESCRIPTION
**Pull request by bwanglzu**
Goals:

- In Finetuner, we allow user to `encode` Documents by batches. When length of DA is large, say 100k, it's easier to have the progress bar natively builtin docarray to monitor the progress.

see it lively:

[Image from Gyazo](https://gyazo.com/9a60e439f95e17fb01203e7509a3fafe)

usage:

```diff
import time
from docarray import Document, DocumentArray

da = DocumentArray()
for i in range(1, 100000):
    da.append(Document(text=str(i)))

print('append finished')

- for batch in da.batch(500):
+ for batch in da.batch(500, show_progress=True):
     time.sleep(0.1)

```

### CODE REVIEW

**Comment by JoanFM** [comment]
> Goals:
> 
> * In Finetuner, we allow user to `encode` Documents by batches. When length of DA is large, say 100k, it's easier to have the progress bar natively builtin docarray to monitor the progress.

Can u please show an example of how it looks before and after the change?

**Review by JoanFM — CHANGES_REQUESTED** [review]
Please, share example of how it looks before and after

**Comment by bwanglzu** [comment]
@JoanFM added in issue

**Review by JoanFM — APPROVED** [review]

File: docarray/array/mixins/group.py [hunk 1]
```diff
@@ -51,8 +52,10 @@ def batch(
 
         :param batch_size: Size of each generated batch (except the last one, which might be smaller, default: 32)
         :param shuffle: If set, shuffle the Documents before dividing into minibatches.
+        :param show_progress: if to show a progress bar on batching.
```
Comment (alaeddine-13) [inline docarray/array/mixins/group.py:55]:
```suggestion
        :param show_progress: if set, show a progress bar when batching documents.
```

File: docarray/array/mixins/group.py [hunk 2]
```diff
@@ -64,7 +67,9 @@ def batch(
         if shuffle:
             random.shuffle(ix)
 
-        for i in range(n_batches):
+        for i in track(
+            range(n_batches), description='Batching', disable=not show_progress
```
Comment (alaeddine-13) [inline docarray/array/mixins/group.py:71]:
```suggestion
            range(n_batches), description='Batching documents', disable=not show_progress
```

**Review by alaeddine-13 — APPROVED** [review]

**Pull request merged**